### ISSUE TITLE
Swatches are not loading

### ISSUE THREAD

**Issue by ghost** [wigman/AjaxSwatches#10]
Attempted to test the newest 0.4.0 release on a custom theme and rwd. So far only getting a colorless swatch with the class 'swatch-link-92' on each product that would normally have swatches present.

**Comment by wigman** [comment]
Hi,

Have you removed all files from app/code/local? Because the last version are located in community. Old files in local could override the new files.
Are you getting any console errors?

**Comment by wigman** [comment]
/app/code/local/wigman ofcourse:) not the entire local folder;)

**Comment by HRR1337** [comment]
Same issue here. Removed the /local/wigman. Though still one swatch loading.

**Comment by wigman** [comment]
Test link anyone? -> willem@e-tailors.nl

**Comment by wigman** [comment]
I got a test-link from HRR1337, where I see the problem.
Jupiter are you also using the Ultimo theme? Might be theme-related.

I need to get in the back-end of one of your sites, or at least check var/system.log for errors and report back.

**Comment by ghost** [comment]
No I tested on rwd and a fully custom theme.

**Comment by wigman** [comment]
Problem solved thanks to HRR1337!

The fix is in app/code/community/wigman/ajaxswatches/helper/mediafallback.php starting on line 101, replace with: 

```
                // using default value as key unless store-specific label is present
                $optionLabel = $optionLabels[$optionId][0]['label'];
                $sortId = $optionLabels[$optionId][0]['sort_id'];
                if (isset($optionLabels[$optionId][$storeId])) {
                    $optionLabel = $optionLabels[$optionId][$storeId]['label'];
                    $sortId = $optionLabels[$optionId][$storeId]['sort_id'];
                }

                // initialize arrays if not present
                if (!isset($optionLabel) || !isset($mapping[$optionLabel])) {
                    $mapping[$optionLabel] = array(
                        'product_ids' => array(),
                    );
                } else {
                    $mapping[$optionLabel] = array();
                }

                $mapping[$optionLabel]['product_ids'][] = $childProduct->getId();
                $mapping[$optionLabel]['label'] = $optionLabel;
                $mapping[$optionLabel]['default_label'] = $optionLabels[$optionId][0]['label'];
                $mapping[$optionLabel]['labels'] = $optionLabels[$optionId];
                $mapping[$optionLabel]['sort_id'] = $sortId;
```

I'm uploading the fix later today.

**Comment by ghost** [comment]
Yea, works properly now, gj.

**Comment by wigman** [comment]
Fixed in version 0.4.1. Thanks for the input!
